benchmark/cublas/blas2/symv.cpp

```diff
@@ -0,0 +1,196 @@
+/**************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename symv.cpp
+ *
+ **************************************************************************/
+
+#include "../utils.hpp"
+
+template <typename scalar_t>
+std::string get_name(std::string uplo, int n) {
+  std::ostringstream str{};
+  str << "BM_Symv<" << blas_benchmark::utils::get_type_name<scalar_t>() << ">/"
+      << uplo << "/" << n;
+  return str.str();
+}
+
+template <typename scalar_t, typename... args_t>
+static inline void cublas_routine(args_t&&... args) {
+  if constexpr (std::is_same_v<scalar_t, float>) {
+    CUBLAS_CHECK(cublasSsymv(std::forward<args_t>(args)...));
+  } else if constexpr (std::is_same_v<scalar_t, double>) {
+    CUBLAS_CHECK(cublasDsymv(std::forward<args_t>(args)...));
+  }
+  return;
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, cublasHandle_t* cuda_handle_ptr,
+         std::string uplo, index_t n, scalar_t alpha, scalar_t beta,
+         bool* success) {
+  // Standard test setup.
+  const char* uplo_str = uplo.c_str();
+
+  index_t incX = 1;
+  index_t incY = 1;
+
+  index_t lda = n;
+  index_t xlen = n;
+  index_t ylen = n;
+
+  // The counters are double. We convert n to double to avoid
+  // integer overflows for n_fl_ops and bytes_processed
+  double n_d = static_cast<double>(n);
+
+  state.counters["n"] = n_d;
+
+  // Compute the number of A non-zero elements.
+  const double A_validVal = (n_d * (n_d + 1) / 2);
+
+  {
+    double nflops_AtimesX = 2 * n_d;
```

Comment:
Did you rather mean `2*A_validVal` here? I believe it's a `2*n_d*n_d` because even-though only half the values are being **used**(read), it's still a full symmetrical matrix-vector operation... But yes for the memory accesses it's indeed half the count.

Reply:
No, that is my mistake. I meant `2*n_d*n_d` but forgot one. Thank you for spotting it!